lib/querystring.js

```diff
@@ -236,6 +240,11 @@ function stringify(obj, sep, eq, options) {
     (encode === qsEscape ? encodeStringified : encodeStringifiedCustom);
 
   if (obj !== null && typeof obj === 'object') {
+    // eslint-disable-next-line node-core/prefer-primordials
+    if (FunctionPrototypeSymbolHasInstance(Map, obj)) {
+      obj = ObjectFromEntries(MapPrototypeEntries(obj));
```

Comment:
I don't think it'd very useful to add support for `Map` objects only to pass them to `Object.fromEntries` (otherwise, why support only `Map` and not all the other objects `Object.fromEntries` supports?). Instead we should implement it in a way that'd be more efficient than user-land (and optionally more resilient)